x-pack/plugins/endpoint/public/embeddables/resolver/store/middleware.ts

```diff
@@ -39,53 +39,34 @@ export const resolverMiddlewareFactory: MiddlewareFactory = context => {
-          const uniquePid = action.payload.selectedEvent?.endgame?.unique_pid;
-          const legacyEndpointID = action.payload.selectedEvent?.agent?.id;
-          [{ lifecycle }, { children }, { events: relatedEvents }] = await Promise.all([
-            context.services.http.get(`/api/endpoint/resolver/${uniquePid}`, {
-              query: { legacyEndpointID },
-            }),
-            context.services.http.get(`/api/endpoint/resolver/${uniquePid}/children`, {
-              query: { legacyEndpointID },
-            }),
-            context.services.http.get(`/api/endpoint/resolver/${uniquePid}/related`, {
-              query: { legacyEndpointID },
-            }),
-          ]);
-          childEvents = children.length > 0 ? flattenEvents(children) : [];
-        } else {
-          const uniquePid = action.payload.selectedEvent.process.entity_id;
-          const ppid = action.payload.selectedEvent.process.parent?.entity_id;
-          async function getAncestors(pid: string | undefined) {
-            if (ancestors.length < maxAncestors && pid !== undefined) {
-              const parent = await context?.services.http.get(`/api/endpoint/resolver/${pid}`);
-              ancestors.push(parent.lifecycle[0]);
-              if (parent.lifecycle[0].process?.parent?.entity_id) {
-                await getAncestors(parent.lifecycle[0].process.parent.entity_id);
-              }
-            }
+        try {
+          let lifecycle: ResolverEvent[];
+          let children: [];
+          if (event.isLegacyEvent(action.payload.selectedEvent)) {
+            const uniquePid = action.payload.selectedEvent?.endgame?.unique_pid;
+            const legacyEndpointID = action.payload.selectedEvent?.agent?.id;
+            [{ lifecycle, children }] = await Promise.all([
+              context.services.http.get(`/api/endpoint/resolver/${uniquePid}`, {
+                query: { legacyEndpointID },
+              }),
+            ]);
+          } else {
+            const uniquePid = action.payload.selectedEvent.process.entity_id;
+            [{ lifecycle, children }] = await Promise.all([
+              context.services.http.get(`/api/endpoint/resolver/${uniquePid}`),
+            ]);
           }
-          [{ lifecycle }, { children }, { events: relatedEvents }] = await Promise.all([
-            context.services.http.get(`/api/endpoint/resolver/${uniquePid}`),
-            context.services.http.get(`/api/endpoint/resolver/${uniquePid}/children`),
-            context.services.http.get(`/api/endpoint/resolver/${uniquePid}/related`),
-            getAncestors(ppid),
-          ]);
+          const mappedChildren = extractChildren(children);
+          const response: ResolverEvent[] = [...lifecycle, ...mappedChildren];
+          api.dispatch({
+            type: 'serverReturnedResolverData',
+            payload: { data: { result: { search_results: response } } },
+          });
+        } catch (error) {
```

Comment:
This error could be any runtime error originating in the try. Consider moving the `try...catch` to contain just the http requests.

Reply:
Thoughts on this? @kqualters-elastic

Reply:
If the http request succeeds but something else in this try/catch fails, the response is not something expected and is effectively the same as a network error.